firedrake/adjoint_utils/blocks/function.py

```diff
@@ -204,338 +201,6 @@ def __str__(self):
         return f"assign({rhs_str})"
 
 
-class InterpolateBlock(Block, Backend):
-    r"""
-    Annotates an interpolator.
-
-    Consider the block as f with 1 forward model output ``v``, and inputs ``u``
-    and ``g`` (there can, in principle, be any number of outputs). The adjoint
-    input is ``vhat`` (``uhat`` and ``ghat`` are adjoints to ``u`` and ``v``
-    respectively and are shown for completeness). The downstream block is ``J``
-    which has input ``v``.
-
-    ::
-
-         _             _
-        |J|--<--v--<--|f|--<--u--<--...
-         ¯      |      ¯      |
-               vhat    |     uhat
-                       |
-                        ---<--g--<--...
-                              |
-                             ghat
-
-    (Arrows indicate forward model direction)
-
-    ::
-
-        J : V ⟶ R i.e. J(v) ∈ R ∀ v ∈ V
-
-    Interpolation can operate on an expression which may not be linear in its
-    arguments.
-
-    ::
-
-        f : W × G ⟶ V i.e. f(u, g) ∈ V ∀ u ∈ W and g ∈ G. f = I ∘ expr I :   X
-        ⟶ V i.e. I(;x) ∈ V ∀ x ∈ X.
-                          X is infinite dimensional.
-        expr: W × G ⟶ X i.e. expr(u, g) ∈ X ∀ u ∈ W and g ∈ G.
-
-    Arguments after a semicolon are linear (i.e. operation I is linear)
```

Comment:
@nbouziani  I presume interpolation is now annotated by something else. I presume whichever block covers assembly? If so, is annotation still correct for interpolating expressions which are not linear, as described here?

Reply:
I think as long as [this test](https://github.com/firedrakeproject/firedrake/blob/c568bfc4f932e72a434cebc74eacb982e02cfd86/tests/regression/test_adjoint_operators.py#L346) passes then yes it does